aion-etl/src/main/java/aion/dashboard/service/DBService.java

```diff
@@ -100,7 +100,7 @@ public void generateCirculatingSupply() throws SQLException {
                     psInsertCirculatingSupply.setString(2, addTotalBlockReward.toString());
                     circulationSupply = (addTotalBlockReward
                             .add(networkBalanceAlloc))
-                            .divide(new BigDecimal(Math.pow(10, 18)));
+                            .divide(new BigDecimal(Math.pow(10, 18))).setScale(18, BigDecimal.ROUND_HALF_EVEN);
```

Comment:
These are functions that are storing the calculated circulating supply values into our database right? If so we don't need these. The idea is we store at maximum precision (I believe we decided `decimal(32,18)`, and when we need to serve it to clients that need less precision we can chop it off then (like you did in the other repository).

Reply:
Yes, also mysql will chop it if has more than 18 decimals, using the scale we will control the round method, I can change it to the suggested one  (RoundingMode.HALF_EVEN) or let MySQL to make the round?

Reply:
Given that 18 decimal places is the max rounding errors is fine. I believe in the kernel we would have used a FLOOR rounding algorithm ( as we were working with BigIntegers).

Reply:
But not a big deal, I don't think we do any division anyways